syntax = "proto3";
package falcon;

///=============================================
/// Enumerations
///=============================================

enum service_id
{
    SYSINFO_SVC_ID      = 0;    // service for getting system information about device
}

enum command_id
{
    CMD_REG_ID          = 0;
    CMD_DEREG_ID        = 1;
    CMD_SET_ID          = 2;
    CMD_GET_ID          = 3;
}

///=============================================
/// Protocol main structures (REQ/RESP)
///=============================================

message request_t
{
    service_id svc_id   = 1;
    command_id cmd_id   = 2;
    string branch       = 3;
    string payload      = 4;
}

message response_t
{
    service_id svc_id   = 1;
    command_id cmd_id   = 2;
    string branch       = 3;
    uint32 status       = 4;
    string resp         = 5;
}

message reg_data_t
{
    string svc_url      = 1;
}

message dereg_data_t
{
    string svc_url      = 1;
}
